plugin/evm/vm.go

```diff
@@ -1439,6 +1454,50 @@ func (vm *VM) getAtomicTx(txID ids.ID) (*Tx, Status, uint64, error) {
 	}
 }
 
+// getAtomicTxFromProcessingBlocks takes in a [txID] and searches for a matching atomic tx in processing blocks
+// Specifically we look from blocks with lastAcceptedBlockHeight + 1  to preferredBlockHeight.
+// This function assumes that we have already searched accepted blocks for atomic txs.
+// We return an error if we cannot find a matching atomic tx in the processing blocks.
+func (vm *VM) getAtomicTxFromProcessingBlocks(txID ids.ID) (*Tx, uint64, error) {
+	return getAtomicTxFromProcessingBlocks(
+		txID,
+		vm.blockChain.CurrentBlock().Number.Uint64(),
+		vm.blockChain.LastAcceptedBlock().Header().Number.Uint64(),
```

Comment:
I think we need to use the vm (plugin/evm) LastAccepted block, since the chain accepts block asynchronously and the VM may have accepted a block the chain has not accepted yet.

Reply:
👀

Reply:
>  I suppose if acceptorQueue is backlogged.
yes this is correct. the vm last accepted height is >= the chain last accepted height, due to the possible lag in the async queue

Reply:
Thanks using the vm's